src/core/task.ts

```diff
@@ -19,7 +20,9 @@ export async function schedule(task: TaskFunction): Promise<void> {
         queue.push({task, resolve});
     });
 
-    if (active === null) { requestAnimationFrame(run); }
+    // If the task queue is empty, invoke the first task in the queue synchronously
+    // This also ensures we don't yield the thread during unload event
+    if (active === null) { run(); }
```

Comment:
>null [](start = 19, length = 4)

Why does active===null mean the same things as queue.lenght == 0? From the comment the latter condition is what I would expect you check.

Reply:
The comment was confusing. I've updated it to reflect that no active task is running. Queue won't be empty at this time because just inserted a task in the queue. Fixed the comment.